### ISSUE TITLE
MetaCommit

### ISSUE THREAD

**Issue by vauvenal5** [vauvenal5/pieShare#25]
Before starting the actual sync of a requested file the node which requested the file needs to send a MetaCommit to inform all other nodes who has been chosen as tracker. This is necessary so all torrent clients work over the same tracker for this file together.

Important is also that every MetaCommit is handled and for every received commit a sharing instance is started. This is needed for the very rare case where two nodes actually accomplish sending a MetaCommit for the same file at the same time which is theoretically possible but should be very rare. This solution to the problem should be given the rarity of this case the most efficient and avoids unnecessary difficult negotiations between the nodes which are requesting.

Also the case where a client which is in process of receiving a file gets a second MetaCommit for the same file should be handled somehow.

Why should not one client simply inform the other about the already ongoing sync?
Because we cant guaranty consistency between the nodes. If the two Commits are send at the exact same time it can happen that different nodes get different Commits first and start sharing before anybody realizes that there is a sync already initialized somewhere else.

We of course could implement some kind of reply voting algorithm. Maybe something like every possible tracker replies with a "ok" or "nok" to every MetaCommit and the one with more "ok"'s gets initialized while the other has to switch the meta file but this seems unnecessarily complicated and is A not needed now and B can be easily added later on top of this simple commit strategy if needed. 

Another case which has to considered is the reaction of FileRequests while a MetaCommit is still not everywhere. Here again I tend to implementing a best effort strategy where nodes which allready have received the MetaCommit send a according response about an already ongoing sync.

The resulting downside is that it could happen that there are two sync circles in the same cluster for the same file. However the big benefit is that every node can still work without the need to rely on response from specific nodes.
